producer/src/main/kotlin/streams/kafka/AdminServiceFactory.kt

```diff
@@ -0,0 +1,13 @@
+package streams.kafka
+
+import org.neo4j.logging.Log
+
+class AdminServiceFactory(private val props: KafkaConfiguration, private val allTopics: List<String>, private val log: Log) {
+
+    fun getAdminService(adminClientApiEnabled: Boolean): AdminService? {
+        return when(adminClientApiEnabled) {
+            false -> DefaultAdminService(log)
+            true -> KafkaAdminService(props, allTopics, log)
+        }
+    }
+}
```

Comment:
no immaginavo qualcosa del tipo
```suggestion
object AdminServiceFactory {
    fun <AdminService> getInstance(props: KafkaConfiguration, allTopics: List<String>, log: Log): AdminService = when (props.adminClientApiEnabled) {
        true -> KafkaAdminService(props, allTopics, log)
        else -> DefaultAdminService(log)
    }
}
```